### ISSUE TITLE
Bug: client.refresh() not working.

### ISSUE THREAD

**Issue by siddiquiaffan** [open-wa/wa-automate-nodejs#2291]
### Are you using the latest version of the library?

- [X] I have checked and am using the latest version of the library.

### Current Behavior

I'm trying to refresh my client using client.refresh() but it's not working and throwing some error.

The error I got is:
```
- session_REFRESH
- Refreshing session
- Opening session in new tab
- Waiting for conflict to close stale tab...
- Checking if fresh session is authenticated...
- Injected new session...
- Downloading patches from https://cdn.openwa.dev/patches.json
- Generating patch hash
- Downloaded patches in 0.563s
- Installing patches
(node:19) UnhandledPromiseRejectionWarning: Error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'sendMessage')
    at __puppeteer_evaluation_script__:1:117221
    at __puppeteer_evaluation_script__:1:450370
    at ExecutionContext._evaluateInternal (/app/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:175:23)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async ExecutionContext.evaluate (/app/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:110:16)
    at async Promise.all (index 0)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:19) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:19) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

### Expected Behavior

It should refresh the client.

### Steps To Reproduce

In my project there are two bots working simultaneously.
i) WhatsApp Bot (wa-automate-nodejs)
ii) Telegram Bot( gramjs)

I've added a command to to telegram bot named `/restart` to refresh the client. When user hits the `/retsart` command a function named `restart` is called.

Restart Function: 
```js
async function restart() {
        log('Restarting WhatsApp Client');
        await client.refresh();
        return {status: 'ok'}
}
```

### Mode

My own code

### create() code

```markdown
const client = await wa.create({useChrome: true, headless: true, authTimeout: 0, chromiumArgs: ['--no-sandbox', '--disable-setuid-sandbox', '--disable-gpu']});
```

### DEBUG INFO

```markdown
{
    WA_VERSION: '2.2144.10',
    PAGE_UA: 'WhatsApp/2.2108.8 Mozilla/5.0 (Macintosh; Intel Mac OS X 11_2_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.90 Safari/537.36',
    WA_AUTOMATE_VERSION: '4.23.7',
    BROWSER_VERSION: 'HeadlessChrome/93.0.4577.63',
    OS: 'Linux 4.9,
    START_TS: 1636782632653
}
```

### Environment

```markdown
- OS: Deployed on railway.app
- Node: 14
- npm:
```

**Comment by smashah** [comment]
I have a feeling this is related to wapi being injected way too early

Might need to reimplement ripe session detection

https://docs.openwa.dev/interfaces/api_model_config.ConfigObject.html#waitForRipeSession

**Comment by AffanTheBest** [comment]
Using ``waitForRipeSession`` can reduce this issue?

And yeah it's ``WAPI`` issue cuz once I got error something like ``WAPI undefined``

**Comment by AffanTheBest** [comment]
So what I need to do basically to solve this issue?


> I have a feeling this is related to wapi being injected way too early
> Might need to reimplement ripe session detection
> https://docs.openwa.dev/interfaces/api_model_config.ConfigObject.html#waitForRipeSession

**Comment by AffanTheBest** [comment]
Any updates?

**Comment by smashah** [comment]
@github-actions run

<details>
<summary>⚡ Release!  ⚡</summary>

```js
(async () => {
function exec(cmd) {
  console.log(execSync(cmd).toString());
}
//set the version type
process.env.VERS = "patch"

// Config
const gitUserEmail = "github-actions[bot]@users.noreply.github.com";
const gitUserName = "github-actions[bot]";

exec(`echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc`);
exec(`git config --global user.email "${gitUserEmail}"`);
exec(`git config --global user.name "${gitUserName}"`);
exec(`npm i -D`);
exec(`npm run release-ci $VERS`);

// types only package
exec('npm config set //registry.npmjs.org/:_authToken $NPM_TOKEN')
exec(`cd ./types-only && npm version $VERS && npm run build && npm publish && cd ..`);
exec(`git commit -a -m 'updated types-only package'`);
exec(`git push --force`);

//comment on the issue
var result = execSync(`npx auto-changelog -o ./tempchangelog.txt --commit-limit false --template ./compact-keepachangelog.hbs --stdout`).toString();

    await postComment(result);

//create changelog image
exec(`npm run release-image`);
exec(`git commit -a -m 'updated release-image'`);
exec(`git push --force`);
  })();
```
</details>

**Comment by smashah** [comment]
# Changelog

## 🚀 Release [4.23.11](https://github.com/open-wa/wa-automate-nodejs/compare/4.23.10...4.23.11) (2021-11-16)

- Release 4.23.11 [`0f07016`](https://github.com/open-wa/wa-automate-nodejs/commit/0f07016d88bf9f07ef21b9fe6c31bbc39ff9daa9)
- 🚑  fix: `waitForRipeSession` #2291 [`594583d`](https://github.com/open-wa/wa-automate-nodejs/commit/594583dd942e1e8d058c8caf8f0294b6436b73b2)
- updated types-only package [`9899e8b`](https://github.com/open-wa/wa-automate-nodejs/commit/9899e8bdfd602923edb409886b58ada04045c967)

**Comment by smashah** [comment]
@AffanTheBest update to the latest version and try again thanks

**Comment by smashah** [comment]
@AffanTheBest to be honest I am unable to replicate this locally. Please add more context to this issue + try to replicate it locally with the easy api.

**Comment by AffanTheBest** [comment]
It's still throwing same error. Even on my local machine.

See below SS:

![Logs of restart function starts here](https://user-images.githubusercontent.com/61529008/142140337-b25fffa1-b687-410a-a300-369135ed4a3f.png)

**Comment by smashah** [comment]
@AffanTheBest 

1. Run `npx @open-wa/wa-automate --headful`
2. Open the api explorer and execute a post request to the `/refresh` endpoint.
3. If the issue continues then make a screen recording of the process

**Comment by AffanTheBest** [comment]
Nope I didn't get any error when used api

**Comment by smashah** [comment]
@AffanTheBest ok then it seems to be related to your implementation.

Please add more context (e.g the config) to this issue.

When you add the necessary context to replicate, I will reopen the issue

**Comment by smashah** [comment]
#2312 

This is fixed in 4.23.17

thanks

**Comment by siddiquiaffan** [comment]
Woah thanks!
